docs/rfc/active/000-ml_models/README.md

```diff
@@ -0,0 +1,123 @@
+- Metrics and Limitations: A model’s performance can be measured in countless ways, but we need to catalog the metrics that are most relevant and useful. Similarly there is a need to store a model's potential limitations that are most useful to quantify.
+- Ensure comparability across models in a well-informed way: Modeling metadata of ML models allows us to compare candidate models' results across not only traditional evaluation metrics but also along the axes of ethical, inclusive, and fairness
+considerations.
+- Promote reproducibility: Often a model is trained on transformed data, there are some preprocessing steps involved in transforming the data e.g. centering, scaling, dealing with missing values, etc. These transforms should be stored as part of model metadata to ensure reproducibility.
+- Ensure Data Governance: Increasing public concern over consumer privacy is resulting in new data laws, such as GDPR and CCPA, causing enterprises to strengthen their data governance & compliance efforts. Therefore, there is a need to store compliance information of ML models containing PII or condidential data (through manual tags or automated process) to eliminate the risk of sensitive data exposure.
+
+## Detailed design
+![high level design](high_level_design.png)
+
+
+As shown in the above diagram, machine learning models are using machine learning features as inputs. These machine learning features
+could be shared across different machine learning models. In the example sketched above, `ML_Feature_1` and `ML_Feature_2` are used as inputs for `ML_Model_A` while `ML_Feature_2`, `ML_Feature_3` and `ML_Feature_4` are inputs for `ML_Model_B`.
+
+### URN Representation
+We'll define two [URNs](../../../what/urn.md): `MLModelUrn` and `MLFeatureUrn`.
+These URNs should allow for unique identification of machine learning models and features, respectively. Machine learning models, like datasets, will be identified by combination of standardized platform urn, name of the model and the fabric type where the model belongs to or where it was generated. 
+
+A machine learning model URN will look like below:
+```
+urn:li:mlModel:(<<platform>>,<<modelName>>,<<fabric>>)
+```
+A machine learning feature will be uniquely identified by it's name and the namespace this feature belongs to.
+A machine learning feature URN will look like below:
+```
+urn:li:mlFeature:(<<namespace>>,<<featureName>>)
+```
+
+### Entities
+There will be 2 top level GMA [entities](../../../what/entity.md) in the design: ML models and ML features.
+It's important to make ML features as a top level entity because ML features could be shared between different ML models.
+
+### ML Model metadata
+- Model properties: Basic information about the ML model
```

Comment:
For model reproducibility, should the training pipeline code which is used to train this specific model be linked as well?